answerking_app/views/category_views.py

```diff
@@ -50,37 +52,37 @@ def post(self, request, *args, **kwargs):
                 status=400,
             )
 
-        response = CategorySerializer(created_category).data
+        response: ReturnDict = CategorySerializer(created_category).data
 
         return JsonResponse(response, status=200, encoder=DjangoJSONEncoder, safe=False)
 
 
 class CategoryDetailView(View):
     @csrf_exempt
-    def get(self, request, *args, **kwargs):
-        category = category_service.get_by_id(kwargs["cat_id"])
+    def get(self, request: HttpRequest, *args, **kwargs) -> JsonResponse:
+        category: Category | None = category_service.get_by_id(kwargs["cat_id"])
 
         if not category:
             return JsonResponse(
                 {"error": {"message": "Request failed", "details": "Object not found"}},
```

Comment:
Can we have a this error messages typed as I previously mentioned to you on slack?

Because I use them in tests too and if we type them we will avoid magic strings, I defined them as
```
class ErrorMessageContent(TypedDict):
    message: str
    details: str


class ErrorMessage(TypedDict):
    error: ErrorMessageContent
```

then you could just change these to
```
error_msg: ErrorMessage= {"error": {"message": "Request failed", "details": "Object not found"}}
return JsonResponse(error_msg)
```

What do you think?

Reply:
but this will cause errors with my code because you have not merged your branch with commit

So this should be raised as another ticket once you have merged your branch with develop no?

Reply:
No, because I wanted to import it from views as it would make more sense to be defined there anyway, I can push mine and you can use `git cherry pick COMMITHASHHERE` to get that commit in your branch

Reply:
Can we have this comment on the main chat please @PietroConvalleAD I agree the error handling is primitive and might need some tweaking depending on people's thoughts